### ISSUE TITLE
[Linux] Really slow framerate

### ISSUE THREAD

**Issue by astroty** [ValveSoftware/Dota-2#782]
Processor Information:

```
Vendor:  GenuineIntel
CPU Family:  0x6
CPU Model:  0x3f
CPU Stepping:  0x2
CPU Type:  0x0
Speed:  4200 Mhz
16 logical processors
8 physical processors
HyperThreading:  Supported
FCMOV:  Supported
SSE2:  Supported
SSE3:  Supported
SSSE3:  Supported
SSE4a:  Unsupported
SSE41:  Supported
SSE42:  Supported
```

Network Information:
    Network Speed:  

Operating System Version:

```
"Manjaro Linux" (64 bit)
Kernel Name:  Linux
Kernel Version:  4.5.0-1-MANJARO
X Server Vendor:  The X.Org Foundation
X Server Release:  11704000
X Window Manager:  KWin
Steam Runtime Version:  <Runtime disabled>
```

Video Card:

```
Driver:  NVIDIA Corporation GeForce GTX 980/PCIe/SSE2

Driver Version:  4.5.0 NVIDIA 361.28
OpenGL Version: 4.5
Desktop Color Depth: 24 bits per pixel
Monitor Refresh Rate: 74 Hz
VendorID:  0x10de
DeviceID:  0x13c0
Number of Monitors:  1
Number of Logical Video Cards:  1
Primary Display Resolution:  1920 x 1080
Desktop Resolution: 1920 x 1080
Primary Display Size: 20,08" x 11,42"  (23,07" diag)
                                        51,0cm x 29,0cm  (58,6cm diag)
Primary Bus: PCI Express 16x
Primary VRAM: 4096 MB
Supported MSAA Modes:  2x 4x 8x 16x 
```

Sound card:

```
Audio device: AV200
```

Memory:

```
RAM:  32052 Mb
```

Miscellaneous:

```
UI Language:  English
LANG:  pt_BR.UTF-8
Microphone:  Not set
Total Hard Disk Space Available:  1840048 Mb
Largest Free Hard Disk Block:  593097 Mb
VR Headset: None detected
```

Installed software:

I'm getting around 70-140fps on linux no matter what setting or resolution I use, while on Windows (OpenGL) I get 200+ consistently.
I've tried lots of others DE like XFCE (Which showed improvement) and LXDE. But the performance wasn't that much affected by it.
I also tested other WM to no result
Tried booting through other systems like Ubuntu and the improvement was comparable, even though better. Tried with a clean install of Manjaro XFCE and it showed same improvement of Ubuntu.
Tried with latest Nvidia driver and the 352.79 with same (kinda worse) results)

**Comment by astroty** [comment]
Setting __GL_THREADED_OPTIMIZATIONS=0 gave me a huge performance boost and stability. I can get around 170fps now consistently, without much variation. Maybe 20 less at most.

**Comment by astroty** [comment]
Checking around my system I found an script for kwin in /etc/profile.d/ export __GL_YIELD="USLEEP". Disabling this and enabling the __GL_THREADED_OPTIMIZATIONS raised my fps to 130-200+
Those two settings might have been causing incompatibility, probably giving me a lower frame rate. The frame rate variates too much though, but I guess the framerate being slow to a crawl issue was resolved (partially. I think windows still get a sightly higher framerate without that much variation)
All in all, I'll be marking this issue resolved for the time being.
